src/Infrastructure/LeanCode.ForceUpdate/LeanCode.ForceUpdate.Services/CQRS/VersionSupportQH.cs

```diff
@@ -0,0 +1,53 @@
+using LeanCode.CQRS.Execution;
+using LeanCode.ForceUpdate.Contracts;
+using Microsoft.AspNetCore.Http;
+using VersionSupport = LeanCode.ForceUpdate.Contracts.VersionSupport;
+
+namespace LeanCode.ForceUpdate.Services.CQRS;
+
+public class VersionSupportQH : IQueryHandler<VersionSupport, VersionSupportDTO?>
+{
+    private readonly IOSVersionsConfiguration iOSConfiguration;
+    private readonly AndroidVersionsConfiguration androidConfiguration;
+    private readonly VersionHandler versionHandler;
+
+    public VersionSupportQH(
+        IOSVersionsConfiguration iOSConfiguration,
+        AndroidVersionsConfiguration androidConfiguration,
+        VersionHandler versionHandler
+    )
+    {
+        this.iOSConfiguration = iOSConfiguration;
+        this.androidConfiguration = androidConfiguration;
+        this.versionHandler = versionHandler;
+    }
+
+    public Task<VersionSupportDTO?> ExecuteAsync(HttpContext context, VersionSupport query)
+    {
+        if (!Version.TryParse(query.Version, out var version) || !Enum.IsDefined<PlatformDTO>(query.Platform))
+        {
+            return Task.FromResult<VersionSupportDTO?>(null);
+        }
+
+        var result = query.Platform switch
+        {
+            PlatformDTO.IOS
+                => versionHandler.CheckVersion(
+                    version,
+                    iOSConfiguration.MinimumRequiredVersion,
+                    iOSConfiguration.CurrentlySupportedVersion,
+                    context
+                ),
+            PlatformDTO.Android
+                => versionHandler.CheckVersion(
+                    version,
+                    androidConfiguration.MinimumRequiredVersion,
+                    androidConfiguration.CurrentlySupportedVersion,
+                    context
+                ),
+            _ => throw new InvalidOperationException("Invalid platform"),
```

Comment:
Throwing an exception on invalid input is just going to confuse clients if they get 500 in response when it's really their fault. How about logging an error _that includes the invalid value we got_ and returning `null` instead?

Reply:
It will never go there as it's checked in `if (!Version.TryParse(query.Version, out var version) || !Enum.IsDefined<PlatformDTO>(query.Platform))` added log there.

Reply:
Ah, the fact that it's a proper `InvalidOperationException` instead of something like `UnreachableException` led me the believe that path was actually reachable. My bad.